### PR TITLE
feat: add some info about connected wifi

### PR DESCRIPTION
**Pull request by rubiin**
Adds some stats similar to waybar on wifi

image

Totally opt in
image

### CODE REVIEW

**Comment by Jas-SinghFSU** [comment]
There are scenarios where the AGS network service returns:
1. Name as an empty string `""`
2. Strength as `-1` indicating the lack of a valid strength (think null)
3. Frequency as `-1` indicating the lack of a valid frequency

We should handle all these scenarios where if these values are captured, we display `--` instead. So if no wifi is connected:
```
Network: None
Signal Strength: --
Frequency: --
```
Or something similar. Essentially we should handle invalid values.

**Comment by Jas-SinghFSU** [comment]
We still need to handle the above scenarios. In case that you're not connected to a Wifi, it will show a negative value for the signal and frequency. We can instead format it like this when invalid values are present:
```
Network: None
Signal Strength: --
Frequency: --
```

**Comment by rubiin** [comment]
sure thing , will update that asap

**Comment by rubiin** [comment]
@Jas-SinghFSU added fallback

File: modules/menus/network/wifi/WirelessAPs.ts [hunk 1]
```diff
@@ -178,35 +178,77 @@ const renderWAPs = (
...
-                                                .catch((err) => {
-                                                    connecting.value = '';
-                                                    console.error(`Error while forgetting "${ap.ssid}": ${err}`);
+                            child: Widget.Box({
+                                children: [
+                                    Widget.Button({
+                                        class_name: 'menu-icon-button network',
+                                        child: Widget.Label({
+                                            tooltip_text: 'Disconnect',
+                                            class_name: 'menu-icon-button network disconnect',
```
Comment (Jas-SinghFSU) [inline modules/menus/network/wifi/WirelessAPs.ts:187]:
```suggestion
                                            class_name: 'menu-icon-button network disconnect txt-icon',
```
We always have to assign txt-icon class to Label based nerdfont icons so they align properly :smile:

File: modules/menus/network/wifi/WirelessAPs.ts [hunk 2]
```diff
@@ -178,35 +178,77 @@ const renderWAPs = (
...
-
-                                            Utils.execAsync(`nmcli connection delete ${connectionId} "${ap.ssid}"`)
-                                                .then(() => (connecting.value = ''))
-                                                .catch((err) => {
-                                                    connecting.value = '';
-                                                    console.error(`Error while forgetting "${ap.ssid}": ${err}`);
+                            child: Widget.Box({
+                                children: [
+                                    Widget.Button({
+                                        class_name: 'menu-icon-button network',
```
Comment (Jas-SinghFSU) [inline modules/menus/network/wifi/WirelessAPs.ts:184]:
```suggestion
                                        class_name: 'menu-icon-button network disconnect',
```

File: modules/menus/network/wifi/WirelessAPs.ts (hunk 1 above)
Comment (rubiin) [reply modules/menus/network/wifi/WirelessAPs.ts:187]: :D I forgot to add that in a rush

File: modules/bar/network/index.ts [hunk 3]
```diff
@@ -68,7 +79,8 @@ const Network = (): BarBoxChild => {
                             }
                             return Widget.Label({
                                 class_name: 'bar-button-label network-label',
-                                label: wfi.ssid ? `${trunc ? wfi.ssid.substring(0, tSize) : wfi.ssid}` : '--',
+                                label: wfi.ssid ? `${trunc ? wfi.ssid.substring(0, tSize) : wfi.ssid}` : '',
```
Comment (Jas-SinghFSU) [inline modules/bar/network/index.ts:82]:
```suggestion
                                label: wfi.ssid ? `${trunc ? wfi.ssid.substring(0, tSize) : wfi.ssid}` : '--',
```

**Review by Jas-SinghFSU — APPROVED** [review]

**Comment by Jas-SinghFSU** [comment]
Awesome stuff :heart:

**Pull request merged**